plugins/GSdx/GSDevice.cpp

```diff
@@ -133,9 +133,9 @@ void GSDevice::Present(GSTexture* sTex, GSTexture* dTex, const GSVector4& dRect,
 
 GSTexture* GSDevice::FetchSurface(int type, int w, int h, bool msaa, int format)
 {
-	GSVector2i size(w, h);
+	const GSVector2i size(w, h);
 
-	for(list<GSTexture*>::iterator i = m_pool.begin(); i != m_pool.end(); ++i)
+	for(std::deque<GSTexture*>::const_iterator i = m_pool.cbegin(); i != m_pool.cend(); ++i)
```

Comment:
`deque` itself instead of `list` is for sure an improvement in linear scan speed (though erase at random position can be slower depending on the implementation).
Maybe it could be worth it to use array lookup + `vector` scan to squeeze additional speed in `FetchSurface` and `Recycle`.
I don't get why can't we keep track of the VRAM available instead of hardcoding maximum sizes though both in `Recycle` and `AgePool`.